python/paddle/v2/framework/framework.py

```diff
@@ -436,11 +436,13 @@ def block(self, index):
     def current_block(self):
         return self.blocks[self.current_block_idx]
 
-    def append_backward(self, target, no_grad_set):
+    def append_backward(self, target, no_grad_set=None):
         """
         return map(param_name -> (grad_name, block_index, op_index))
         """
         assert isinstance(target, Variable)
+        if no_grad_set is None:
+            no_grad_set = set()
```

Comment:
I think `append_backward(self, target, no_grad_set=None):` => `append_backward(self, target, no_grad_set=set())` will do the same trick?

Reply:
Well, I think maybe sometimes users will write `program.append_backward(target=var, no_grad_set=None)`

Reply:
@tonyyang-svail 

The default parameter in Python has a huge difference with other languages.  It is mutable in Python.

See http://effbot.org/zone/default-values.htm

Reply:
I see. Thanks for the heads up.